pytorch_lightning/callbacks/model_summary.py

```diff
@@ -50,14 +50,16 @@ def __init__(self, max_depth: Optional[int] = 1):
         self._max_depth: int = max_depth
 
     def on_pretrain_routine_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
-        if trainer.is_global_zero and self._max_depth is not None and not trainer.testing:
```

Comment:
reviewer: the check for testing gets removed here. it is redundant as we are in the "on_pretrain_routine_start" hook